tools/gfx/command-encoder-com-forward.h

```diff
@@ -1,124 +1,142 @@
-    {                                                                                                   \
-        ResourceCommandEncoderBase::resolveResource(                                                    \
-            source, sourceState, sourceRange, dest, destState, destRange);                              \
-    }                                                                                                   \
-    virtual SLANG_NO_THROW void SLANG_MCALL resolveQuery(                                               \
-        IQueryPool* queryPool,                                                                          \
-        GfxIndex index,                                                                                 \
-        GfxCount count,                                                                                 \
-        IBufferResource* buffer,                                                                        \
-        Offset offset) override                                                                         \
-    {                                                                                                   \
-        ResourceCommandEncoderBase::resolveQuery(queryPool, index, count, buffer, offset);              \
-    }                                                                                                   \
-    virtual SLANG_NO_THROW void SLANG_MCALL writeTimestamp(IQueryPool* pool, GfxIndex index)            \
-        override                                                                                        \
-    {                                                                                                   \
-        ResourceCommandEncoderBase::writeTimestamp(pool, index);                                        \
-    }                                                                                                   \
-    virtual SLANG_NO_THROW void SLANG_MCALL beginDebugEvent(const char* name, float rgbColor[3])        \
-        override                                                                                        \
-    {                                                                                                   \
-        ResourceCommandEncoderBase::beginDebugEvent(name, rgbColor);                                    \
-    }                                                                                                   \
-    virtual SLANG_NO_THROW void SLANG_MCALL endDebugEvent() override                                    \
-    {                                                                                                   \
-        ResourceCommandEncoderBase::endDebugEvent();                                                    \
+#define SLANG_GFX_FORWARD_RESOURCE_COMMAND_ENCODER_IMPL(ResourceCommandEncoderBase)                \
+    virtual SLANG_NO_THROW SlangResult SLANG_MCALL queryInterface(                                 \
```

Comment:
We might want to consider some x macro magic to generate this kind of code. That code can simplify the definition to a x macro that has the return type, the name and the param types and generates all the other stuff from that.

Reply:
Yeah that would be a nice clean up.